index.js

```diff
@@ -15,149 +36,72 @@ const LIFE360_CLIENT_SECRET = "cFJFcXVnYWJSZXRyZTRFc3RldGhlcnVmcmVQdW1hbUV4dWNyR
  * countryCode:  Optional phone country code, defaults to 1 if not specified.
  * returns:      Life360 session.
  */
-module.exports.authenticate = function (username, password, phone, countryCode) {
-  return new Promise((resolve, reject) => {
-    countryCode = typeof countryCode !== 'undefined' ? countryCode : 1;
+module.exports.authenticate = function (username, password) {
     username = typeof username !== 'undefined' ? username : '';
-    phone = typeof phone !== 'undefined' ? phone : '';
 
-    if (!password) throw new Error("No password specified.");
+    if (!password) return Promise.reject(new Error("No password specified."))
 
-    const LIFE360_LOGIN_URL = "https://api.life360.com/v3/oauth2/token.json";
-    const LIFE360_LOGIN_POSTDATA = `countryCode=${countryCode}&username=${username}&phone=${phone}&password=${password}&grant_type=password`;
-
-    const options = {
-      url: LIFE360_LOGIN_URL,
-      method: 'POST',
-      body: LIFE360_LOGIN_POSTDATA,
-      headers: {
-        'Authorization': `Authorization: Basic ${LIFE360_CLIENT_SECRET}`,
-        'Content-Type': 'application/x-www-form-urlencoded'
-      },
-      json: true
+    const form = {
+        username,
+        password,
+        grant_type: 'password'
     };
-    // console.log(`Logging in as user: ${username}, password: ${password}`);
-    request(options)
-      .then(response => {
-        if (response.statusCode != 200) {
-          reject(new Error(`Life360 server error logging in: ${response.statusCode}`));
-        }
-        response = response.body;
-        if (!response['access_token']) throw new Error("Unauthorized");
-        const session = {
-          access_token: response['access_token'],
-          token_type: response['token_type']
+    const bodyFormData = Object.keys(form).reduce((formData, key) => {
+        formData.append(key, form[key]);
+        return formData;
+    },  new FormData())
```

Comment:
Wonder if this will solve #2.

Reply:
Might be, will try to test with a password, containing "+" sign.